js/speckled_moth.js

```diff
@@ -1,239 +1,267 @@
-  newVec = newVec.multiply(.5)
-  var r = gaussianBoxMuller(0,geneSpread)
-  var g = gaussianBoxMuller(0,geneSpread)
-  var b = gaussianBoxMuller(0,geneSpread)
-  var mutation = $V([r,g,b])
-  var newColor = newVec.add(mutation).map( function(e){return Math.floor(e%256)
-  })
-  return newColor
-};
-
-Moth.prototype.chooseDirection = function(testX, testY){
-  var vR = Math.random()*5
-  var twoPi = 2* 3.141592
-  var thet = twoPi*Math.random()
-  this.vX = testX || vR*Math.cos(thet)
-  this.vY = testY || vR*Math.sin(thet)
-};
-
-Moth.prototype.moveStep = function(){
-  this.pX = this.pX + this.vX
-  this.pY = this.pY + this.vY
-  this.reflect()
-  this.view.update()
-};
-
-Moth.prototype.reflect =function(){
-  var bound = this.field.size - this.diameter
-  if(this.pX < this.diameter/2){
-    // console.log("reflection!")
-    this.vX = -this.vX
-    this.pX = -this.pX
-  }else if(this.pX - bound > this.diameter/2){
-    // console.log("reflection!")
-    this.vX = -this.vX
-    this.pX = bound - (this.pX-bound)*2
-  }else if(this.pY < this.diameter/2){
-    // console.log("reflection!")
-    this.vY = -this.vY
-    this.pY = -this.pY
-  }else if(this.pY - bound > this.diameter/2){
-    // console.log("reflection!")
-    this.vY = -this.vY
-    this.pY = bound - (this.pY-bound)*2
-  };
-};
-
-Moth.prototype.die = function(){
-  this.timeOfDeath = Date.now();
-  this.$el.off()
-  this.$el.hide()
-};
-
-// ============ end object definitions
+    args = merge(defaults, args)
+    this.field = args["field"]
+    this.idNum = args["idNum"];
+    this.color = args["colorVec"];
+    this.pX = Math.random()*this.field.size;
+    this.pY = Math.random()*this.field.size;
+    this.vX = 0;
+    this.vY = 0;
+    this.timeOfDeath = 0
+    this.diameter = 17;
+    var divID = 'moth_'+this.idNum;
+    this.$el = $("<div id='"+divID+"'></div>");  
+    this.view = mothView(this)
+  }
+
+  Moth.prototype.mate = function(otherMoth, geneSpread){
+    // why not this?
+    var newVec = this.color.add(otherMoth.color).multiply(.5);
+    var r = gaussianBoxMuller(0,geneSpread);
+    var g = gaussianBoxMuller(0,geneSpread);
+    var b = gaussianBoxMuller(0,geneSpread);
+    var mutation = $V([r,g,b]);
+    return newVec.add(mutation).map(this.floor256);
+  };
+
+  Moth.prototype.floor256 = function(give_me_a_better_name){
+    return Math.floor(give_me_a_better_name % 256);
+  };
+
+  Moth.prototype.chooseDirection = function(testX, testY){
+    var vR = Math.random()*5
+    var twoPi = 2* 3.141592
+    var thet = twoPi*Math.random()
+    this.vX = testX || vR*Math.cos(thet)
+    this.vY = testY || vR*Math.sin(thet)
+  };
+
+  Moth.prototype.moveStep = function(){
+    this.pX = this.pX + this.vX
+    this.pY = this.pY + this.vY
+    this.reflect()
+    this.view.update()
+  };
+
+  Moth.prototype.reflect =function(){
+    var bound = this.field.size - this.diameter
+    if(this.pX < this.diameter/2){
+      // console.log("reflection!")
```

Comment:
Consider removing the commented out `console.log` - this usually stands out to me as a need for a test.